src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +98,68 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        // $$$
+        internal static TAttribute GetAttr<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            TAttribute attributeSource = parameter.GetCustomAttribute<TAttribute>(inherit: false);
+
+            // Walk hierachy.
+            // It's possible attribute is not on the parameter but is still on the something 
+            // higher up in the hiearchy
+            ICustomAttributeProvider provider = parameter;
+
+            while (provider != null)
+            {
+                var attrs = provider.GetCustomAttributes(typeof(Attribute), false);
+                foreach (var attr in attrs)
+                {
+                    if (attributeSource == null)
+                    {
+                        attributeSource = attr as TAttribute;
+                        continue;
+                    }
+
+                    if (attr.GetType().IsAssignableFrom(typeof(TAttribute)))
+                    {
+                        // Merge properties. Give precedence to "closest" one. 
+                        foreach (var prop in attr.GetType().GetProperties(BindingFlags.Instance | BindingFlags.Public))
+                        {
+                            if (prop.CanRead && prop.CanWrite)
+                            {
+                                var currentValue = prop.GetValue(attributeSource);
+                                if (currentValue == null)
+                                {
+                                    var value = prop.GetValue(attr);
+                                    prop.SetValue(attributeSource, value);
```

Comment:
We have to merge the properties together into a single attr.   Say we have:
   [Blob] [Storage(account="X"] 
We need to set Blob.account = x